beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestAddonChangeAlignment.java

```diff
@@ -0,0 +1,192 @@
+import org.killbill.billing.catalog.api.PlanPhaseSpecifier;
+import org.killbill.billing.entitlement.api.DefaultEntitlementSpecifier;
+import org.killbill.billing.entitlement.api.Subscription;
+import org.killbill.billing.platform.api.KillbillConfigSource;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class TestAddonChangeAlignment extends TestIntegrationBase {
+
+    @Override
+    protected KillbillConfigSource getConfigSource(final Map<String, String> extraProperties) {
+        final Map<String, String> allExtraProperties = new HashMap<>(extraProperties);
+        allExtraProperties.put("org.killbill.catalog.uri", "catalogs/testAddonChangeAlignment");
+        return super.getConfigSource(null, allExtraProperties);
+    }
+
+    @Test(groups = "slow")
+    public void testChangeAlignmentStartOfBundleEOTChange() throws Exception {
+        final LocalDate today = new LocalDate(2023, 8, 1);
+        clock.setDay(today);
+
+        final Account account = createAccountWithNonOsgiPaymentMethod(getAccountData(1));
+
+        //create base 2023-08-01
+        final PlanPhaseSpecifier spec = new PlanPhaseSpecifier("basic-monthly");
+        busHandler.pushExpectedEvents(NextEvent.CREATE, NextEvent.BLOCK, NextEvent.INVOICE);
+        final UUID baseEntId = entitlementApi.createBaseEntitlement(account.getId(), new DefaultEntitlementSpecifier(spec), null, null, null, false, true, Collections.emptyList(), callContext);
+        assertListenerStatus();
+
+        //Base CTD is 2023-08-01 as expected
+        Subscription baseSub = subscriptionApi.getSubscriptionForEntitlementId(baseEntId, false, callContext);
+        Assert.assertEquals(baseSub.getChargedThroughDate(), new LocalDate(2023, 8, 1));
+
+        clock.setDay(new LocalDate(2023, 8, 5));
+
+        //create addon 2023-08-05
+        final PlanPhaseSpecifier addonSpec = new PlanPhaseSpecifier("BasicAO-monthly");
+        busHandler.pushExpectedEvents(NextEvent.CREATE, NextEvent.BLOCK, NextEvent.INVOICE);
+        final UUID addonEntId = entitlementApi.addEntitlement(baseSub.getBundleId(), new DefaultEntitlementSpecifier(addonSpec), null, null, false, Collections.emptyList(), callContext);
+        assertListenerStatus();
+
+        //ADDON CTD is 2023-08-05 as expected
+        Subscription addonSub = subscriptionApi.getSubscriptionForEntitlementId(addonEntId, false, callContext);
+        Assert.assertEquals(addonSub.getChargedThroughDate(), new LocalDate(2023, 8, 05));
+
+        clock.setDay(new LocalDate(2023, 8, 15));
+
+        //change addon with start of bundle alignment on 2023-08-15 - change will happen immediately as subscription is in TRIAL phase
+        busHandler.pushExpectedEvents(NextEvent.CHANGE, NextEvent.INVOICE);
+        final PlanPhaseSpecifier newAddOnSpec = new PlanPhaseSpecifier("BasicAOStartOfBundle-monthly");
+        addonSub.changePlan(new DefaultEntitlementSpecifier(newAddOnSpec), Collections.emptyList(), callContext);
```

Comment:
It's a little bit confusing to follow:  We started with `BasicAO-monthly` (which has a `START_OF_BUNDLE` alignment as this is the default case) and then we move to `BasicAOStartOfBundle-monthly` (which also has a `START_OF_BUNDLE` alignment, explicitly specified).

1. Is the goal to test a change of plan from 2 products both using `START_OF_BUNDLE` alignment?
2. If so, could you modify the naming/catalog as those 2 products have the same behavior (`START_OF_BUNDLE` alignments), but one uses default behavior and it is not reflected in the name while the second uses custom rule and has it specified in the name.

Reply:
> Is the goal to test a change of plan from 2 products both using START_OF_BUNDLE alignment?

The goal is to test a change of plan to a product that uses `START_OF_BUNDLE` alignment irrespective of the default case.  I've now also renamed the products in the catalog. Let me know if this is helpful.